### PR TITLE
[ticket/16397] Expose user identifier

### PR DESCRIPTION
**Pull request by mrgoldy**
PHPBB3-16397

Checklist:

- [x] Correct branch: master for new features; 3.3.x & 3.2.x for fixes
- [x] Tests pass
- [x] Code follows coding guidelines: [master](https://area51.phpbb.com/docs/dev/master/development/coding_guidelines.html), [3.3.x](https://area51.phpbb.com/docs/dev/3.3.x/development/coding_guidelines.html) and [3.2.x](https://area51.phpbb.com/docs/dev/3.2.x/development/coding_guidelines.html)
- [x] Commit follows commit message [format](https://area51.phpbb.com/docs/dev/3.3.x/development/git.html)

Tracker ticket (set the ticket ID to **your ticket ID**):

https://tracker.phpbb.com/browse/PHPBB3-16397

### CODE REVIEW

File: phpBB/phpbb/session.php [hunk 1]
```diff
@@ -25,6 +25,7 @@ class session
 	var $forwarded_for = '';
 	var $host = '';
 	var $session_id = '';
+	public $id = ANONYMOUS;
```
Comment (marc1706) [inline phpBB/phpbb/session.php]: Design-wise I think it makes more sense to have this default to an invalid value, i.e. 0.

Comment (mrgoldy) [reply phpBB/phpbb/session.php]: Usually I would completely agree, you set base variables to 'nothing' (`null`, `''`, `0`, etc).
However, considering the amount of checks that are done with `$user_id <> ANONYMOUS` I would stick with this. Cause if this is set to `0` and somehow not changed, all of a sudden in various places the user is not considered a guest. Should never really happen but still.

Comment (rxu) [reply phpBB/phpbb/session.php:28]: Now we have issues with this default assignment in installer (fixed [here](https://github.com/phpbb/phpbb/commit/09e308a06d7fcaa624afe7e4a86e178bab84a362#diff-8bfd43635044b97335b9c309cd081a1a)) and in console [here](https://github.com/phpbb/phpbb/pull/5846#discussion_r436789271).

Comment (marc1706) [reply phpBB/phpbb/session.php:28]: Yep, ran into this while testing for 3.3.1-RC1.

**Pull request merged**